src/components/NewCommentForm.tsx

```diff
@@ -75,25 +210,40 @@
             id="comment-body"
             name="body"
             placeholder="Type comment here"
-            className="textarea is-danger"
+            className={classnames('textarea', {
+              'is-danger': formState.errors.body,
+            })}
+            value={formState.formData.body}
+            onChange={handleChange}
           />
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Enter some text
-        </p>
+        {formState.errors.body && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Enter some text
+          </p>
+        )}
       </div>
 
       <div className="field is-grouped">
         <div className="control">
-          <button type="submit" className="button is-link is-loading">
+          <button
+            type="submit"
+            className={classnames('button is-link', {
+              'is-loading': isLoading,
```

Comment:
Great use of the `classnames` library to conditionally apply classes based on the form state.